src/manifest.rs

```diff
@@ -229,13 +241,24 @@ impl str::FromStr for Manifest {
         let mut parser = toml::Parser::new(&input);
 
         parser.parse()
-              .ok_or(parser.errors.pop())
+              .ok_or(format_parse_error(parser))
               .map_err(Option::unwrap)
               .map_err(From::from)
               .map(|data| Manifest { data: data })
     }
 }
 
+fn format_parse_error(mut parser: toml::Parser) -> Option<ManifestError> {
+    match parser.errors.pop() {
+        Some(error) => {
```

Comment:
Since you are only really concerning yourself with the `Some(error)` case, I think the code would look cleaner using `parser.errors.pop().map(|error| …)` instead of the match. (It's only a question of style, it does the exact same thing.)